modules/core/src/lib/layer.js

```diff
@@ -864,6 +864,26 @@ export default class Layer extends Component {
     }
   }
 
+  updateAutoHighlight(info) {
+    if (this.props.autoHighlight) {
+      this._updateAutoHighlight(info);
+    }
+  }
+
+  // May be overriden by classes
+  _updateAutoHighlight(info) {
```

Comment:
Nit: I have to admit that I find this pair of functions with and without underscore somewhat confusing. I wonder if there could be a naming scheme that makes things clearer. 

I think at some point the private method naming across the `Layer` class could stand a review. I have heard external users complain about it being hard to follow the logic.

Reply:
We have several pairs like this in the Layer class, e.g. `updateAttributes` and `_updateAttributes`. I agree that a review would help clarify things.

Reply:
This PR looks great to me in terms of architecture. I've also faced a couple of times with the issue it tries to solve. I've to say I felt the same confusion with the naming of `_updateAutoHighlight`and `updateAutoHighlight`.

However, I've to admit I find nothing better